command/CommandController.js

```diff
@@ -737,12 +743,29 @@ async function searchForPocketArticleByUrl(getBody, url, extendedData) {
       if (keysArr.length > 0) {
         let isArticleId = 0;
 
-        // store the id for first object that is an article
-        while (
-          isArticleId < keysArr.length &&
-          jsonBody.list[keysArr[isArticleId]].is_article != '1'
-        ) {
-          isArticleId++;
+        if (theItemId) {
+          // We have the item ID, so try and match to resolved id
+          while (
+            isArticleId < keysArr.length &&
+            jsonBody.list[keysArr[isArticleId]].resolved_id != theItemId
+          ) {
+            logger.debug(
+              'ResolvedId is: ' +
+                jsonBody.list[keysArr[isArticleId]].resolved_id
+            );
+            isArticleId++;
+          }
+          logger.debug(
+            'Final ResolvedId is: ' +
+              jsonBody.list[keysArr[isArticleId]].resolved_id
+          );
+        } else {
+          while (
+            isArticleId < keysArr.length &&
+            jsonBody.list[keysArr[isArticleId]].is_article != '1'
+          ) {
+            isArticleId++;
+          }
         }
 
         if (isArticleId < keysArr.length) {
```

Comment:
If you go with the above `findIndex` then this if statement needs an update to:
```
        if (isArticleId >= 0 && isArticleId < keysArr.length) {
```
since the `isArticleId` value will be -1 if the item_id is not found in keysArr.

Reply:
changed.  Good catch. thanks.